### PR TITLE
[WIP] Bug 1439730 Make Secrets have multiple views

### PR DESCRIPTION
**Pull request by JQuinnie**
🚧  **Currently Work-In-Progress, will update when finalized** 👷‍♀️ 

[Bug 1439730](https://bugzilla.mozilla.org/show_bug.cgi?id=1439730)
Taskcluster [Secrets page](http://tools.taskcluster.net/secrets) currently has two-panel view layout with a secret list on the left and the create/edit panel on the right, see _before_ image below. The team is trying to move away from that design to a one-panel view, see _after_ image below.

**Before View**
taskclusterbeforecapture

**After View**
taskclusteraftercapture

### CODE REVIEW

File: src/components/SecretEditor/index.js [hunk 1]
```diff
@@ -4,12 +4,12 @@ import { safeLoad, safeDump } from 'js-yaml';
 import { Alert, Button, ButtonToolbar, Glyphicon } from 'react-bootstrap';
 import { fromNow } from 'taskcluster-client-web';
 import Icon from 'react-fontawesome';
-import Spinner from '../../components/Spinner';
```
Comment (eliperelman) [inline src/components/SecretEditor/index.js]: Can I ask why we moved this file to a component? It seems very specific to the view in question, so I am not sure how re-usable it is in this context.

Comment (JQuinnie) [reply src/components/SecretEditor/index.js]: Hi Eli,

Originally thinking to follow the example of [scopeInspector](https://github.com/taskcluster/taskcluster-tools/tree/master/src/views/ScopeInspector) and have something similar to [other editor components](https://github.com/taskcluster/taskcluster-tools/blob/794a4b2bfdaab754b68e68982c4a47e57d49c018/src/views/ScopeInspector/ScopeInspector.js#L17-L18) in the component files... I still do have a few clean ups in regards to naming conventions, but I am open to your advice if you would like me to keep it in the same view folder.

**Comment by JQuinnie** [comment]
Hi Eli,
I was planning to provide some screenshots of the progress, will do that later today 😁

**Comment by eliperelman** [comment]
I'm visiting this page and it appears that it thinks I am creating a secret:

https://deploy-preview-463--taskcluster-web.netlify.com/secrets/garbage%2Fmarco%2Ftest-code-coverage-secret

screen shot 2018-03-01 at 7 30 01 am

@JQuinnie could you have a look at that? Thank you!

**Comment by JQuinnie** [comment]
Hi Eli,

In regards to your [link above](https://deploy-preview-463--taskcluster-web.netlify.com/secrets/garbage%2Fmarco%2Ftest-code-coverage-secret)  which is the issue I was facing and reached out in [BugAhoy](https://bugzilla.mozilla.org/show_bug.cgi?id=1439730)... I will take your suggestions there, but I think the thing I'm missing is just passing the 'secret' when you click on the link... in between views, the object is to either create secret or edit... your screenshot shows that the 'secret' was not passed therefore not able to render to the edit function.

I am still working it.

**Comment by JQuinnie** [comment]
Please refer to PR #479

**Pull request closed without merging**